src/ts/content/extensions/enforceAuthAndTagIndex.ts

```diff
@@ -0,0 +1,37 @@
+import {ConfigKey} from "../../common/entities/config";
+import config from "../../common/entities/config";
+import {onBackgroundEvent, onceBackgroundEvent} from "../util/onBackgroundEvent";
+import {BackgroundEvent} from "../../common/backgroundEvent";
+import {isAuthorized} from "./author/util/isAuthorized";
+import {createModal} from "../../common/ui/modal";
+import {UIColour} from "../../common/ui/colour";
+
+const isSpreadSheetLinkSet = async () => config.get(ConfigKey.SpreadsheetLink);
+
+window.addEventListener("pageshow", async () => {
+	if (await config.get(ConfigKey.EnforceAuthAndTagIndex)) {
+		if (!(await isAuthorized()) || !(await isSpreadSheetLinkSet())) {
```

Comment:
I _think_  you're recreating `onLogged`.
`onLoggedIn` is called when both of these conditions are met (like when you invoke exit)
and `onLoggedOut` is called if one of the conditions aren't met (when you invoke exit)

If you use `onLogged`, you also get the advantage that the modal would pop up again if you log out in another tab

Reply:
Just a thought, but not necessary -- if we're going to make things react to actions in other places like with onLogged, you _could_ even add `onBackgroundEvent(EnableEnforceAccess)` or something, that re-triggers this pageshow callback

```typescript
const callbackWithABetterName = () => {
	// ... ur code
};

onBackgroundEvent(BackgroundEvent.Access, callbackWithABetterName);
window.addEventListener("pageshow", callbackWithABetterName);
```

Reply:
hmmm but the downside is that `onLoggedOut` will get called multiple times........ which would mean multiple overlays. Let's leave it and think about it.

Might want to consider logic deduplicating in the future